test/puppetlabs/puppetdb/testutils/mem.clj

```diff
@@ -0,0 +1,27 @@
+(ns puppetlabs.puppetdb.testutils.mem
+  (:require [clojure.java.shell :as shell])
+  (:import [java.lang.management ManagementFactory]
+           [java.nio.file Files NoSuchFileException]))
+
+(def getpid
+  "Returns the integer PID of the current process."
+  (let [pid (atom nil)]
+    (fn []
+      (if-let [p @pid]
+        p
+        (reset! pid (->> (ManagementFactory/getRuntimeMXBean)
+                         .getName
+                         (re-find #"[0-9]+")
+                         Integer/parseInt))))))
+
+(defn dump-heap [filename]
```

Comment:
Yes, that's intentional - the commented commands just show how we can use dump-heap (when we're trying to diagnose problems), but we don't need it for the test, which is just checking to see that PDB doesn't die if we try to send something approaching max-command-size through.